spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/SpannerReadConverter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.core.convert;
+
+import org.springframework.data.convert.CustomConversions;
+
+/**
+ * @author Balint Pato
+ */
+public class SpannerReadConverter extends AbstractSpannerCustomConverter {
+
+	public SpannerReadConverter(CustomConversions customConversions) {
+		super(customConversions, null);
+	}
+
+	@Override
+	public <T> T convert(Object sourceValue, Class<T> targetType) {
```

Comment:
It looks like the only difference between this method and the parent version is that the parent does a check if the desired type is the same as that of source rather than here if it is a child type. I think this difference is more an error and  we should have done the parent method the way you have done this. 

We should change the parent version to do the check for child type rather than equals-type. With that it doesn't seem like this class has a reason to exist anymore right?

Reply:
Oh, yeah I wanted to tackle that, thanks for catching it!